engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -238,7 +239,8 @@ public void onReturnValue(VirtualFrame frame, Object result) {
           if (cachedResult != null) {
             throw context.createUnwind(cachedResult);
           }
-        } else if (node instanceof ExpressionNode expressionNode) {
+        } else if (node instanceof ExpressionNode expressionNode
+            && !(result instanceof UnresolvedConstructor)) {
```

Comment:
I don't like this condition either, but...

> disable the instrumentation for UnresolvedConstructors

`UnresolvedConstructor` is a _value_. There is no instrumentation for values, only _code - e.g. nodes_ - get instrumented. `ExpressionNode` doesn't know what _value_ flows thru it in advance. It can be `Atom`, its constructor or `DataflowError` or `UnresolvedConstructor`... as such `ExpressionNode` cannot decide (in advance) to _not instrument_ based on a value.

The filtering could [happen in ProgramExecutionSupport](https://github.com/enso-org/enso/blob/dcccad9f7b795579be1f65ba19db048cc2acf8fb/engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala#L435), if you think it is better... but it has to happen somewhere in the `IdExecution` instrument or later and _not in the instrumentation_.

Reply:
Let's leave it this way then